src/mlpack/tests/init_rules_test.cpp

```diff
@@ -123,4 +127,37 @@ BOOST_AUTO_TEST_CASE(OivsInitTest)
   BOOST_REQUIRE_EQUAL(1, 1);
 }
 
+// Test the GaussianInitialization class.
+BOOST_AUTO_TEST_CASE(GaussianInitTest)
+{
+  arma::mat weights;
+  arma::cube weights3d;
+  const size_t row = 7;
+  const size_t col = 7;
+  const size_t slice = 2;
+  double mean, mean3d;
+  double var,var3d;
+  mean=mean3d=1;
+  var=var3d=1;
+  GaussianInitialization t(0, 0.2);
+  RandomSeed(21);
```

Comment:
No, this won't. I did try it without a seed. But the problem is until and unless we increase the number of iterations to a very high value we can guarantee that the mean would be arbitrarily close to epsilon value. Thats why i have used a seed here.